src/main/java/com/hotel/mappers/book/request/GuestMapper.java

```diff
@@ -0,0 +1,53 @@
+package com.hotel.mappers.book.request;
+
+import com.hotel.service.common.Address;
+import com.hotel.service.common.UserInfo;
+import org.opentravel.ota._2003._05.*;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GuestMapper {
+
+    public ArrayOfResGuestsTypeResGuest map(UserInfo userInfo) {
+        ArrayOfResGuestsTypeResGuest arrayOfResGuestsTypeResGuest = new ArrayOfResGuestsTypeResGuest();
+        ArrayOfResGuestsTypeResGuest.ResGuest resGuest = new ArrayOfResGuestsTypeResGuest.ResGuest();
+        ArrayOfProfilesTypeProfileInfo profiles = new ArrayOfProfilesTypeProfileInfo()  ;
+        resGuest.setProfiles(profiles);
```

Comment:
profiles.getProfileInfo().add(profileInfo) is done in last. So if we do  resGuest.setProfiles(profiles) before that , does really the value will get set inside profiles?